testing/run.sh

```diff
@@ -197,13 +192,36 @@ while getopts "${CMDOPTS}" opt; do
   esac
 done
 
-if ((INITCPIO)) ; then
-  GENZBM_FLAGS+=( "-i" )
+if ! ((INITCPIO)) && ! ((DRACUT)); then
+  # Prefer mkinitcpio if available
+  if command -v mkinitcpio >/dev/null 2>&1 && [ -d "${TESTDIR}/mkinitcpio.d" ]; then
+    INITCPIO=1
+  else
+    DRACUT=1
+  fi
+elif ((INITCPIO)) && ((DRACUT)); then
+    echo "ERROR: dracut and mkinitcpio are mutually exclusive"
+    exit 1
 fi
 
-# Zero out any customizations from the last run
-: > "${TESTDIR}/dracut.conf.d/testing.conf"
-: > "${TESTDIR}/mkinitcpio.d/testing.conf"
+# Enable logging for dracut and verify the configuration exists
+if ((DRACUT)); then
+  if [ ! -d "${TESTDIR}/dracut.conf.d" ]; then
+    echo "ERROR: cannot use dracut without dracut.conf.d"
+    exit 1
+  fi
+
+  AAPPEND+=( "rd.log=kmsg" )
```

Comment:
Despite the name of `rd.log`, this is actually a `initcpio` [framework parameter.](https://github.com/archlinux/mkinitcpio/blob/master/init_functions#L93-L115)

Reply:
Not sure how I misplaced this, but I relocated it.